src/profile/components/multiItemAddressRow/MultiItemAddressRow.tsx

```diff
@@ -0,0 +1,189 @@
+import EditButtons from '../editButtons/EditButtons';
+import FormikDropdown, {
+  HdsOptionType,
+} from '../../../common/formikDropdown/FormikDropdown';
+import getLanguageCode from '../../../common/helpers/getLanguageCode';
+import LabeledValue from '../../../common/labeledValue/LabeledValue';
+import getCountry from '../../helpers/getCountry';
+import { getFormFields } from '../../helpers/formProperties';
+import SaveIndicator from '../saveIndicator/SaveIndicator';
+import { useCommonEditHandling } from '../../hooks/useCommonEditHandling';
+import { RowItemProps } from '../multiItemEditor/MultiItemEditor';
+
+type FormikValues = AddressValue;
+
+function MultiItemAddressRow(props: RowItemProps): React.ReactElement {
+  const { data, testId, dataType } = props;
+  const value = data.value as AddressValue;
+  const { address, city, postalCode, countryCode } = value;
+  const { t, i18n } = useTranslation();
+  const lang = i18n.languages[0];
+  const applicationLanguage = getLanguageCode(i18n.languages[0]);
+  const countryList = countries.getNames(applicationLanguage);
+  const countryOptions = Object.keys(countryList).map(key => ({
+    value: key,
+    label: countryList[key],
+  }));
+  const formFields = getFormFields(dataType);
+  const {
+    isNew,
+    isEditing,
+    actionHandler,
+    currentAction,
+  } = useCommonEditHandling(props);
+  const { hasFieldError, getFieldErrorMessage } = createFormFieldHelpers<
+    FormikValues
+  >(t, isNew);
+
+  const { primary } = data;
+
+  if (isEditing) {
+    return (
+      <Formik
+        initialValues={{
+          address,
+          city,
+          postalCode,
+          countryCode,
+        }}
+        onSubmit={async values => {
+          await actionHandler('save', values);
+        }}
+        validationSchema={addressSchema}
+      >
+        {(formikProps: FormikProps<FormikValues>) => (
+          <Form className={commonFormStyles.multiItemForm}>
+            <h4 className={commonFormStyles.sectionTitle}>
+              {primary
+                ? t('profileInformation.primaryAddress')
+                : t('profileInformation.address')}
+            </h4>
+            <div className={commonFormStyles.multiItemWrapper}>
+              <Field
+                name="address"
+                id={`${testId}-address`}
+                maxLength={formFields.address.max as number}
+                as={TextInput}
+                invalid={hasFieldError(formikProps, 'address')}
+                aria-invalid={hasFieldError(formikProps, 'address')}
+                helperText={getFieldErrorMessage(formikProps, 'address')}
+                labelText={t(formFields.address.translationKey)}
+                autoFocus
+                aria-labelledby={`${dataType}-address-helper`}
+              />
+              <Field
+                name="postalCode"
+                id={`${testId}-postalCode`}
+                maxLength={formFields.postalCode.max as number}
```

Comment:
Tested this and the maxLength seemed to be 5? I think there might be longer postal codes in other countries, so should we support longer ones as well, because we want to support other countries too?

Reply:
There is no documentation why 5 was selected in old Profile UI. Perhaps, because BE didn't always support more chars. I asked around and now BE supports 32 chars. UX designer approved this too. So will change to 32.

Reply:
Done